File: api-openbis-java/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/property/update/PropertyTypeUpdate.java
/*
 * Copyright ETH 2014 - 2023 Zürich, Scientific IT Services
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.asapi.v3.dto.property.update;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IObjectUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.ListUpdateMapValues;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.ListUpdateValue.ListUpdateAction;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.DataType;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.property.id.IPropertyTypeId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 */
@JsonObject("as.dto.property.update.PropertyTypeUpdate")
public class PropertyTypeUpdate implements IUpdate, IObjectUpdate<IPropertyTypeId>
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private IPropertyTypeId typeId;

    @JsonProperty
    private DataType dataType;
    
    @JsonProperty
    private FieldUpdateValue<String> label = new FieldUpdateValue<String>();

    @JsonProperty
    private FieldUpdateValue<String> description = new FieldUpdateValue<String>();

    @JsonProperty
    private FieldUpdateValue<String> schema = new FieldUpdateValue<String>();

    @JsonProperty
    private FieldUpdateValue<String> transformation = new FieldUpdateValue<String>();
    
    @JsonProperty
    private ListUpdateMapValues metaData = new ListUpdateMapValues();

    @JsonProperty
    private FieldUpdateValue<String> pattern = new FieldUpdateValue<String>();

    @JsonProperty
    private FieldUpdateValue<String> patternType = new FieldUpdateValue<String>();

    @Override
    @JsonIgnore
    public IPropertyTypeId getObjectId()
    {
        return getTypeId();
    }

    @JsonIgnore
    public DataType getDataTypeToBeConverted()
    {
        return dataType;
    }

    @JsonIgnore
    public void convertToDataType(DataType dataType)
    {
        this.dataType = dataType;
    }

    @JsonIgnore
    public IPropertyTypeId getTypeId()
    {
        return typeId;
    }

    @JsonIgnore
    public void setTypeId(IPropertyTypeId typeId)
    {
        this.typeId = typeId;
    }

    @JsonIgnore
    public FieldUpdateValue<String> getLabel()
    {
        return label;
    }

    @JsonIgnore
    public void setLabel(String label)
    {
        this.label.setValue(label);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getDescription()
    {
        return description;
    }

    @JsonIgnore
    public void setDescription(String description)
    {
        this.description.setValue(description);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getSchema()
    {
        return schema;
    }

    @JsonIgnore
    public void setSchema(String schema)
    {
        this.schema.setValue(schema);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getTransformation()
    {
        return transformation;
    }

    @JsonIgnore
    public void setTransformation(String transformation)
    {
        this.transformation.setValue(transformation);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getPattern()
    {
        return pattern;
    }

    @JsonIgnore
    public void setPattern(String pattern)
    {
        this.pattern.setValue(pattern);
    }

    @JsonIgnore
    public FieldUpdateValue<String> getPatternType()
    {
        return patternType;
    }

    @JsonIgnore
    public void setPatternType(String patternType)
    {
        this.patternType.setValue(patternType);
    }

    @JsonIgnore
    public ListUpdateMapValues getMetaData()
    {
        return metaData;
    }

    @JsonIgnore
    public void setMetaDataActions(List<ListUpdateAction<Object>> actions)
    {
        metaData.setActions(actions);
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("typeId", typeId).toString();
    }

}

File: api-openbis-javascript/src/v3/as/dto/property/update/PropertyTypeUpdate.js
define([ "stjs", "as/dto/common/update/FieldUpdateValue", "as/dto/common/update/ListUpdateMapValues" ], function(stjs, FieldUpdateValue, ListUpdateMapValues) {
	var PropertyTypeUpdate = function() {
		this.label = new FieldUpdateValue();
		this.description = new FieldUpdateValue();
		this.schema = new FieldUpdateValue();
		this.transformation = new FieldUpdateValue();
		this.metaData = new ListUpdateMapValues();
	};
	stjs.extend(PropertyTypeUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.property.update.PropertyTypeUpdate';
		constructor.serialVersionUID = 1;
		prototype.typeId = null;
		prototype.dataType = null;
		prototype.label = null;
		prototype.description = null;
		prototype.schema = null;
		prototype.transformation = null;
		prototype.metaData = null;

		prototype.getObjectId = function() {
			return this.getTypeId();
		};
		prototype.getDataTypeToBeConverted = function() {
			return this.dataType;
		};
		prototype.convertToDataType = function(dataType) {
			this.dataType = dataType;
		};
		prototype.getTypeId = function() {
			return this.typeId;
		};
		prototype.setTypeId = function(typeId) {
			this.typeId = typeId;
		};
		prototype.getLabel = function() {
			return this.label;
		};
		prototype.setLabel = function(label) {
			this.label.setValue(label);
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description.setValue(description);
		};
		prototype.getSchema = function() {
			return this.schema;
		};
		prototype.setSchema = function(schema) {
			this.schema.setValue(schema);
		};
		prototype.getTransformation = function() {
			return this.transformation;
		};
		prototype.setTransformation = function(transformation) {
			this.transformation.setValue(transformation);
		};
		prototype.getMetaData = function() {
			return this.metaData;
		};
		prototype.setMetaDataActions = function(actions) {
			this.metaData.setActions(actions);
		};
	}, {
		typeId : "IPropertyTypeId",
		label: {
			name: "FieldUpdateValue",
			arguments: ["String"]
		},
		description: {
			name: "FieldUpdateValue",
			arguments: ["String"]
		},
		schema: {
			name: "FieldUpdateValue",
			arguments: ["String"]
		},
		transformation: {
			name: "FieldUpdateValue",
			arguments: ["String"]
		},
		dataType : "DataType",
		metaData : "ListUpdateMapValues"
	});
	return PropertyTypeUpdate;
})
